packages/shorebird_cli/lib/src/commands/release/macos_releaser.dart

```diff
@@ -0,0 +1,212 @@
+  bool get codesign => argResults['codesign'] == true;
+
+  @override
+  ReleaseType get releaseType => ReleaseType.macos;
+
+  @override
+  Future<void> assertArgsAreValid() async {
+    if (argResults.wasParsed('release-version')) {
+      logger.err(
+        '''
+The "--release-version" flag is only supported for aar and ios-framework releases.
+        
+To change the version of this release, change your app's version in your pubspec.yaml.''',
+      );
+      throw ProcessExit(ExitCode.usage.code);
+    }
+
+    if (argResults.rest.contains('--obfuscate')) {
+      // Obfuscated releases break patching, so we don't support them.
+      // See https://github.com/shorebirdtech/shorebird/issues/1619
+      logger
+        ..err('Shorebird does not currently support obfuscation on macOS.')
+        ..info(
+          '''We hope to support obfuscation in the future. We are tracking this work at ${link(uri: Uri.parse('https://github.com/shorebirdtech/shorebird/issues/1619'))}.''',
+        );
+      throw ProcessExit(ExitCode.unavailable.code);
+    }
+  }
+
+  @override
+  Future<void> assertPreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+        validators: doctor.macosCommandValidators,
+        supportedOperatingSystems: {Platform.macOS},
+      );
+    } on PreconditionFailedException catch (e) {
+      throw ProcessExit(e.exitCode.code);
+    }
+
+    final flutterVersionArg = argResults['flutter-version'] as String?;
+    if (flutterVersionArg != null) {
+      final version =
+          await shorebirdFlutter.resolveFlutterVersion(flutterVersionArg);
+      if (version != null && version < minimumSupportedMacosFlutterVersion) {
+        logger.err(
+          '''
+macOS releases are not supported with Flutter versions older than $minimumSupportedMacosFlutterVersion.
+For more information see: ${supportedFlutterVersionsUrl.toLink()}''',
+        );
+        throw ProcessExit(ExitCode.usage.code);
+      }
+    }
+  }
+
+  @override
+  Future<FileSystemEntity> buildReleaseArtifacts() async {
+    if (!codesign) {
+      logger
+        ..info(
+          '''Building for device with codesigning disabled. You will have to manually codesign before deploying to device.''',
+        )
+        ..warn(
+          '''shorebird preview will not work for releases created with "--no-codesign". However, you can still preview your app by signing the generated .xcarchive in Xcode.''',
+        );
+    }
+
+    // Delete the Shorebird supplement directory if it exists.
+    // This is to ensure that we don't accidentally upload stale artifacts
+    // when building with older versions of Flutter.
+    final shorebirdSupplementDir =
+        artifactManager.getMacosReleaseSupplementDirectory();
+    if (shorebirdSupplementDir?.existsSync() ?? false) {
+      shorebirdSupplementDir!.deleteSync(recursive: true);
+    }
```

Comment:
This shouldn't be needed on MacOS because it was introduced after the supplement directory changes so there will always be a supplement artifact generated by the flutter_tool.